src/main/java/org/broadinstitute/dsm/model/Filter.java

```diff
@@ -268,6 +266,33 @@ else if (filter.getFilter1() != null && filter.getFilter2().getValue() != null &
         return finalQuery;
     }
 
+    /**
+     * Uses the appropriate date converter (if given) to write SQL that can
+     * compare either exact dates or "in the day" dates.
+     */
+    private static String generateDateComparisonSql(Filter filter, DBElement dbElement, String comparison, Object arg) {
+        String column = filter.getColumnName(dbElement);
+        SqlDateConverter dateConverter = dbElement.getDateConverter();
+
+        Instant instant = null;
+        try {
+            instant = new SimpleDateFormat("yyyy-MM-dd").parse(arg.toString()).toInstant().atZone(ZoneId.of("UTC")).toInstant();
```

Comment:
`SimpleDateFormat` is quite old already, I think it is better idea to use date related things from `time` package, for example: 
`LocalDate.parse(arg.toString(), DateTimeFormatter.ISO_LOCAL_DATE).atStartOfDay().toInstant(ZoneOffset.UTC)` should be better.